src/main.rs

```diff
@@ -155,13 +156,28 @@ fn document_path(path: &Path, root: &Path, document: &mut Document, settings: &S
 }
 
 fn document_header(path: &Path, content: &str, document: &mut Document, settings: &Settings) {
-    parse_unreal_cpp_header(content, document, settings).unwrap_or_else(|error| {
-        panic!(
-            "Could not parse Unreal C++ header file content!\nFile: {:?}\nError:\n{}",
-            path,
-            error.to_string()
-        )
-    });
+    match parse_unreal_cpp_header(content, document, settings) {
```

Comment:
i like the idea of this, i would like to test it too - could you give a header snippet that can show best how it prints errors? reply to this comment would be enough, no need to make a test file in changes for it.

Reply:
The log output looks something like this. In fact, I originally hoped to print out the complete rule chain, but I have tried some methods and still can't achieve it, so at present, this is as far as I can go, which seems to be almost the same as the error information.
```
Error in Pest rules chain 'pointer_type -> reference_type -> unpackness' at line                TFunction<void(int)> &&Callback,:    --> 152:17
    |
152 |           TFunction<void(int)> &&Callback,
    |                         ^---
    |
    = expected pointer_type, reference_type, or unpackness
```

Reply:
It seems to be of little use. Shall I revert this part of the code?

Reply:
i am not sure, but if we aim here for providing more useful errors for the user, we might try to catch/inspect an error and print it with PEST's error formatting. it also should be easy to do, we just need to dive deeper and find how to use that formatter.